### PR TITLE
Upgrade Modin to 0.18

### PR DESCRIPTION
**Pull request by LeonLuttenberger**
### Detail
- Upgrade Modin to 0.18.0

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 6fd8a1ee49ed3e283531135adee3ba122b404a93
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=46f5bf0c-a741-4ec1-8484-64d25b90b291%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: 6fd8a1ee49ed3e283531135adee3ba122b404a93
* Result: SUCCEEDED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=daa41e44-d3ad-4119-bce2-b08058ced417%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-ATYtnXPE7MOa
* Commit ID: 6fd8a1ee49ed3e283531135adee3ba122b404a93
* Result: SUCCEEDED
* [Build Logs](https://i1emq028pj.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b2cc2787-345a-4bce-a8ce-7c71ef754c18%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 6fd8a1ee49ed3e283531135adee3ba122b404a93
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e097f4e2-c201-4713-bad7-4a623358a11b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: pyproject.toml [hunk 1]
```diff
@@ -28,7 +28,7 @@ python = ">=3.8, <4.0"
 
 boto3 = "^1.24.11"
 botocore = "^1.27.11"
-pandas = "^1.2.0, <=1.5.1, !=1.5.0" # Exclusion per: https://github.com/aws/aws-sdk-pandas/issues/1678
+pandas = "^1.2.0, !=1.5.0" # Exclusion per: https://github.com/aws/aws-sdk-pandas/issues/1678
```
Comment (jaidisido) [inline pyproject.toml:31]: We ve had problems with breaking changes in minor versions of pandas so we should at least pin it to `< 1.6` perhaps?

File: pyproject.toml [hunk 2]
```diff
@@ -46,7 +46,7 @@ gremlinpython = "^3.5.2"
 backoff = ">=2.0.0,<3.0.0"
 SPARQLWrapper = { version = "^2.0.0", optional = true }
 pyodbc = { version = "~4.0.32", optional = true }
-modin = { version = "~0.16.0", optional = true }
+modin = { version = "~0.18.0", optional = true }
```
Comment (jaidisido) [inline pyproject.toml:49]: Same issue than with Ray, we need to make sure that Glue can support this version first...

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 0e4a0f8f06c460d443fecc70eb42b62936ba91a0
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c287fd10-141f-471a-9956-210210510a40%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: 0e4a0f8f06c460d443fecc70eb42b62936ba91a0
* Result: FAILED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e3f2061a-548e-452e-9d72-d03e2ad38f88%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-ATYtnXPE7MOa
* Commit ID: 0e4a0f8f06c460d443fecc70eb42b62936ba91a0
* Result: SUCCEEDED
* [Build Logs](https://i1emq028pj.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7e8e7097-0436-4343-8de4-db50d101bdd3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 94be1197bd1827203b3c18de249cf35a68e1dc8f
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c04199f0-16e6-4d6f-bfbc-0156f9e62145%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubStandardCodeBuild8C06-llutOAimTATs
* Commit ID: 94be1197bd1827203b3c18de249cf35a68e1dc8f
* Result: SUCCEEDED
* [Build Logs](https://jsv9u8gote.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=17d54791-7580-4e3c-8d40-5b4d23edae03%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubLoadTests5656BB24-ATYtnXPE7MOa
* Commit ID: 94be1197bd1827203b3c18de249cf35a68e1dc8f
* Result: SUCCEEDED
* [Build Logs](https://i1emq028pj.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c5bb6a9a-ceb4-4ebf-b73f-35bf804ab8a1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request closed without merging**